pagespeed/controller/rpc_handler.h

```diff
@@ -228,7 +228,7 @@ void RpcHandler<AsyncService, RequestT, ResponseT>::ReadDone(RefPtrT ref) {
 template <typename AsyncService, typename RequestT, typename ResponseT>
 bool RpcHandler<AsyncService, RequestT, ResponseT>::Finish(
     const ::grpc::Status& status) {
-  if (state_ == FINISHED) {
+  if (state_ == FINISHED || state_ == INIT) {
```

Comment:
I guess that a change in gRPC is making Finish() hang now if the client isn't properly initialized. Probably the call to Finish() is now happening before something that it used to happen after. However, the early exit for INIT here means you're not setting state_ to FINISHED. That's causing the extra call to HandleRequest(), which you definitely don't want. Returning false for this case is fine, but you still need to set state_

Reply:
Thanks a lot for taking a look! and, yeah, that seems correct, what I observed to happen is that ordering seems to have changed. With the latest grpc versions, `InitDone()` is called after `Finish()`. I backed my changes out in favor of a tweak in `InitDone()`.
With that I can retain the original test expectations (except for the grpc status).